### PR TITLE
f-checkout@v3.0.1 Create cookie on submit as guest for GTM.

### PR DESCRIPTION
**Pull request by BanksySan**
GTM requires a cookie adding when a guest user checks out.

### CODE REVIEW

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 1]
```diff
@@ -3303,11 +3303,20 @@ describe('Checkout', () => {
...
+                        const expectedExpireDate = now.setMinutes(now.getMinutes() + 1);
+                        expect(expire).toBeCloseTo(expectedExpireDate, -1);
+                    });
+
                     // Act
                     await wrapper.vm.onFormSubmit();
 
                     // Assert
                     expect(updateMessageSpy).toHaveBeenCalled();
+                    expect($cookies.set).toHaveBeenCalled();
```
Comment (FayeCarter) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js]: Could you change the test description name to match the changes please? Or just add a new test.
And then can you do `toHaveBeenCalledWith` please?

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 2]
```diff
@@ -932,6 +932,10 @@ export default {
             this.setSubmittingState(true);
 
             if (this.isFormValid()) {
+                const now = new Date();
+                if (!this.isLoggedIn) {
+                    this.$cookies.set('je-gtm-event', 'engagement|form_checkout_guest|success', now.setMinutes(now.getMinutes() + 1));
+                }
```
Comment (oliversweb) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue]: Maybe only create variable if you need it ....

```suggestion                
                if (!this.isLoggedIn) {
                    const now = new Date();
                    this.$cookies.set('je-gtm-event', 'engagement|form_checkout_guest|success', now.setMinutes(now.getMinutes() + 1));
                }
```
Or maybe do away with the variable ...

```suggestion                
                if (!this.isLoggedIn) {
                    this.$cookies.set('je-gtm-event', 'engagement|form_checkout_guest|success', new Date(Date.now() + 60000));
                }
```

Comment (BanksySan) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: Used `max-age` instead.

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 3]
```diff
@@ -932,6 +932,9 @@ export default {
             this.setSubmittingState(true);
 
             if (this.isFormValid()) {
+                if (!this.isLoggedIn) {
+                    this.$cookies.set('je-gtm-event', 'engagement|form_checkout_guest|success', { maxAge: 60 });
```
Comment (charliearlie) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue:936]: NIT: Could have these values in constants but I've approved anyway.

**Review by charliearlie — DISMISSED** [review]

**Review by FayeCarter — DISMISSED** [review]

**Review by zlatinivanov-je — DISMISSED** [review]

**Comment by BanksySan** [comment]
Replaced by #1437.

**Pull request closed without merging**